lib/src/main/java/io/github/davidallison/android/sdk/penremote/ButtonEvent.kt

```diff
@@ -33,14 +33,16 @@ package io.github.davidallison.android.sdk.penremote
  * https://developer.samsung.com/galaxy-spen-remote/api-reference/com/samsung/android/sdk/penremote/ButtonEvent.html
  */
 class ButtonEvent(event: SPenEvent) {
+    val action: ButtonAction
+    val timeStamp: Long
+
     init {
-        TODO("set variables from constructor")
+        timeStamp = event.timeStamp
+        action =
+            if (event.values[1].toInt() == 0) ButtonAction.ACTION_DOWN else ButtonAction.ACTION_UP
     }
```

Comment:
code style: Android Studio should say that both variables can be defined inline, rather than in an `init { }` block